### PR TITLE
WIP: SSR

### PR DESCRIPTION
**Pull request by lixiaoyan**
### Features

- [x] Proxy Server
- [x] Mobx
- [ ] Apollo
- [x] styled components
- [x] React.hydrate()
- [ ] Dynamic Import
- [ ] Production Nginx Server

### TODO

- [ ] find a better way to let hot reload work together with caddy

### Bugs

- [x] CLIENT_PORT
- [x] nodemon improve
  - [x] nodemon rs
  - [x] reduce restart times
- [x] babel-plugin-styled-components
- [x] ~~jest.js -> entry/jest.js~~
- [x] remove env command

### Docs

- [x] add caddy to prerequisites

### CODE REVIEW

**Comment by zh99998** [comment]
nodemon 命令行输入 rs，会报caddy端口占用

**Comment by lixiaoyan** [comment]
@zh99998 CLIENT_PORT 没有用到…

**Comment by lixiaoyan** [comment]
@lixiaoyan TODO: babel-plugin-styled-components 应该在 devDependencies / 减少 nodemon 重启次数 / 增加 production proxy server / 添加 caddy 到 docs

**Comment by lixiaoyan** [comment]
@lixiaoyan 移动 jest.js 到 entry/jest.js

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ node_js: node
 script:
   - npm run test
   - npm run report-coverage
-  - npm run build:production
+  - env NODE_ENV=production npm run build
```
Comment (oott123) [inline .travis.yml]: 这个 `env` 没啥用吧

File: package.json [hunk 2]
```diff
@@ -21,20 +33,24 @@
     ]
   },
   "dependencies": {
+    "babel-plugin-styled-components": "^1.1.7",
```
Comment (oott123) [inline package.json]: 话说这个不是 dev deps？我不太懂

File: src/server/HTML.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    sheet: PropTypes.object.isRequired,
+    content: PropTypes.string.isRequired,
+  }
+
+  render() {
+    const { manifest, helmet, sheet, content } = this.props
+    return (
+      <html {...helmet.htmlAttributes.toComponent()}>
+        <head>
+          <meta charSet="UTF-8" />
```
Comment (oott123) [inline src/server/HTML.js:17]: 不应该是`charset="utf-8"` 吗……

**Review by oott123 — COMMENTED** [review]
看不太懂

File: src/server/HTML.js (hunk 3 above)
Comment (oott123) [reply src/server/HTML.js:17]: okay 不改了

**Review by oott123 — APPROVED** [review]

**Pull request merged**